Add Azure Stream Analytics SDK project and code

atpham256 opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
- Added StreamAnalytics folder
- Added csproj and sln files to set up project infrastructure
- Added code generated from Swagger specs found here:
https://github.com/Azure/azure-rest-api-specs/tree/master/arm-streamanalytics
- Bumped Assembly version to 2.0.0.0 to signify breaking changes are
accompanied with this new SDK
- Added test infrastructure and tests cases to test the generated code
for major user scenarios
- Ran tests and copied over the session records for playback

NOTE: Code for the Azure Stream Analytics SDK was previously generated
using Hydra/Hyak. Now that the code is being generated using
AutoRest/Swagger specs, the NuGet Package published that will contain
this code will introduce breaking changes compared to the previous NuGet
Package published. As a result, the version of the SDK was bumped up to
2.0.0.0 from 1.8.3.0.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

atpham256 committed:
Add Azure Stream Analytics SDK project and code

- Added StreamAnalytics folder
- Added csproj and sln files to set up project infrastructure
- Added code generated from Swagger specs found here:
https://github.com/Azure/azure-rest-api-specs/tree/master/arm-streamanalytics
- Bumped Assembly version to 2.0.0.0 to signify breaking changes are
accompanied with this new SDK
- Added test infrastructure and tests cases to test the generated code
for major user scenarios
- Ran tests and copied over the session records for playback

NOTE: Code for the Azure Stream Analytics SDK was previously generated
using Hydra/Hyak. Now that the code is being generated using
AutoRest/Swagger specs, the NuGet Package published that will contain
this code will introduce breaking changes compared to the previous NuGet
Package published. As a result, the version of the SDK was bumped up to
2.0.0.0 from 1.8.3.0.

shahabhijeet commented on src/SDKs/StreamAnalytics/StreamAnalytics.Tests/ScenarioTests/TransformationTests.ScenarioTests.cs:
```diff
@@ -0,0 +1,75 @@
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Management.Resources;
+using Microsoft.Azure.Management.Resources.Models;
```
@atpham256 please use the latest Resource Manager nuget 1.6.0 to record all your tests.
You are using really old Resource Manager 1.1.0. We are asking anyone to use the latest nugets to record all your tests.

atpham256 replied:
@shahabhijeet I believe the test infrastructure is set up similar to other test projects where we reference https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/AzSdk.test.reference.props which contains the Resource Manager NuGet.

Shouldn't that be updated if you want people to use the latest NuGet?

atpham256 replied:
@shahabhijeet I manually added the 1.6.0 NuGet to our test project, updated the tests, re-ran them, and updated the session records.

Let me know if I did anything incorrectly. :)

atpham256 committed:
Merge branch 'psSdkJson6' of https://github.com/Azure/azure-sdk-for-net into psSdkJson6

atpham256 committed:
Use most up-to-date ARM NuGet for tests

Updated Stream Analytics tests to use
Microsoft.Azure.Management.ResourceManager version 1.6.0-preview NuGet
instead of the old Microosft.Azure.Management.Resource version
1.1.0-preview NuGet that it was previously using.

Re-ran all of the tests using the new NuGet and updated the session
records.
